src/cljs/rems/spa.cljs

```diff
@@ -455,8 +456,9 @@
 (defn init! []
   (rf/dispatch-sync [:initialize-db])
   (load-interceptors!)
-  (fetch-translations!)
-  (fetch-theme!)
-  (config/fetch-config!)
-  (hook-browser-navigation!)
-  (mount-components))
+  (-> (p/all [(fetch-translations!)
+              (fetch-theme!)
+              (config/fetch-config!)])
+      (p/finally (fn []
+                   (hook-browser-navigation!)
+                   (mount-components)))))
```

Comment:
Doesn't this prevent any useful loading spinner from showing up?

How about not fetching any of these here after page load but instead already injecting them into the base page on the server side?

Reply:
The spinner will show up when it's loading the page content, which is what typically takes the longest time, for example in the catalogue page.

Reply:
It's an option to switch to loading everything with ajax or injecting everything to the html. Currently it's a mixture of both. I created an idea ticket for it.

Reply:
The slowest part of the initial page load is downloading and parsing the JavaScript, during which no JavaScript-generated spinner can anyways be shown. It would need to be in the static HTML.